rustls/src/client/hs.rs

```diff
@@ -576,6 +572,12 @@ impl State for ExpectServerHello {
         // For TLS1.3, start message encryption using
         // handshake_traffic_secret.
         if conn.common.is_tls13() {
+            if !suite.usable_for_version(TLSv1_3) {
+                return Err(conn
+                    .common
+                    .illegal_param("server chose unusable ciphersuite for version"));
+            }
+
```

Comment:
In the Future we should introduce a similar `Tls13CipherSuite` type that is used for in the TLS 1.3 state machine for similar reasons.